### ISSUE TITLE
[Bug]: updater deleting custom temp directory

### ISSUE THREAD

**Issue by shortywz** [netdata/netdata#18724]
### Bug description

It appears that the issue raised and solved in https://github.com/netdata/netdata/issues/5289 has reappeared in the last few days builds.

/etc/netdata/.environment has

`export NETDATA_TMPDIR_PATH="/root/tmp"`

at the end of netdata-updater.sh, /root/tmp is being deleted resulting in it being unable to run until the directory is recreated manually.

```
Tue Oct  8 09:26:27 CDT 2024 : INFO: netdata-updater.sh:  Checking if a newer version of the updater script is available.
/usr/libexec/netdata/netdata-updater.sh: line 449: /root/tmp/latest-commit.json: No such file or directory
/usr/libexec/netdata/netdata-updater.sh: line 460: /root/tmp/latest-commit.json: No such file or directory
Tue Oct  8 09:26:27 CDT 2024 : INFO: netdata-updater.sh:  Downloading newest version of updater script.
/usr/libexec/netdata/netdata-updater.sh: line 586: cd: /root/tmp: No such file or directory
```

### Expected behavior

custom tmpdir is not deleted after the updater script is ran

### Steps to reproduce

1. set custom tmpdir via /etc/netdata/.environment
2. create tmpdir
3. run updater
4. run updater again

### Installation method

kickstart.sh

### System info

```shell
Linux ###.###.### 4.18.0-553.5.1.el8_10.x86_64 #1 SMP Wed Jun 5 09:12:13 EDT 2024 x86_64 x86_64 x86_64 GNU/Linux
/etc/almalinux-release:AlmaLinux release 8.10 (Cerulean Leopard)
/etc/centos-release:AlmaLinux release 8.10 (Cerulean Leopard)
/etc/os-release:NAME="AlmaLinux"
/etc/os-release:VERSION="8.10 (Cerulean Leopard)"
/etc/os-release:ID="almalinux"
/etc/os-release:ID_LIKE="rhel centos fedora"
/etc/os-release:VERSION_ID="8.10"
/etc/os-release:PLATFORM_ID="platform:el8"
/etc/os-release:PRETTY_NAME="AlmaLinux 8.10 (Cerulean Leopard)"
/etc/os-release:ANSI_COLOR="0;34"
/etc/os-release:LOGO="fedora-logo-icon"
/etc/os-release:CPE_NAME="cpe:/o:almalinux:almalinux:8::baseos"
/etc/os-release:
/etc/os-release:ALMALINUX_MANTISBT_PROJECT="AlmaLinux-8"
/etc/os-release:ALMALINUX_MANTISBT_PROJECT_VERSION="8.10"
/etc/os-release:REDHAT_SUPPORT_PRODUCT="AlmaLinux"
/etc/os-release:REDHAT_SUPPORT_PRODUCT_VERSION="8.10"
/etc/os-release:SUPPORT_END=2029-06-01
/etc/redhat-release:AlmaLinux release 8.10 (Cerulean Leopard)
/etc/system-release:AlmaLinux release 8.10 (Cerulean Leopard)
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.99.0-275-nightly
    Installation Type __________________________________________ : binpkg-rpm
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ :
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 4.18.0-553.5.1.el8_10.x86_64
    Operating System ___________________________________________ : AlmaLinux
    Operating System ID ________________________________________ : almalinux
    Operating System ID Like ___________________________________ : rhel centos fedora
    Operating System Version ___________________________________ : 8.10 (Cerulean Leopard)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 8
    CPU Frequency ______________________________________________ : 3500000000
    RAM Bytes __________________________________________________ : 16341508096
    Disk Capacity ______________________________________________ : 4000808706048
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : NO
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
    libmnl (library for working with netfilter) ________________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by ilyam8** [comment]
Hi, @shortywz. Why did you decide to use `NETDATA_TMPDIR_PATH`? This variable is internal to `netdata-updater.sh` and should not be set outside the script.

I think you need to use `NETDATA_TMPDIR`.

**Comment by Ferroin** [comment]
@ilyam8 is correct, if you want to force a specific temporary directory here, you need to set `NETDATA_TMPDIR` instead.

ON a side note, since you’re using native packages, you don’t actually even _need_ the updater script, all it does for native packages (aside from a bunch of complicated logic to update itself if needed before running the rest of the update) is tell the package manager to update first the Netdata repository configuration packages, and then the Netdata package itself.

**Comment by shortywz** [comment]
Thanks guys - the cron for the updater got added during the installer a few years ago it looks like - I've just gone ahead and removed it since as it was pointed out I don't need it for the native packages.

As to your specific question @ilyam8 - that was the suggested resolution in another ticket where someone had similar issues to mine a number of years back. I've changed it (even though it shouldn't really be necessary anymore).
